.github/workflows/ci.yml

```diff
@@ -41,16 +35,18 @@ jobs:
     - name: Clean to prepare for NSec build
       run: |
         dotnet clean
-    - name: Run tests
-      run: |
-        dotnet test
     - name: Package
       run: |
         cd $GITHUB_WORKSPACE/src/DotNetLightning.Core
         dotnet pack -p:Configuration=Release -p:Version=1.1.0-date`date +%Y%m%d-%H%M`.git-`echo $GITHUB_SHA | cut -c 1-7`
-    - name: Upload nuget
+    - name: Run tests
+      run: |
+        dotnet test
+    - name: Upload nuget packages (BouncyCastle and native)
       run: |
         cd $GITHUB_WORKSPACE/src/DotNetLightning.Core
         if [ ${{ secrets.NUGET_API_KEY }} ] && [ $GITHUB_REF == "refs/heads/master" ]; then
-            dotnet nuget push ./bin/Release/DotNetLightning.Core*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
+            dotnet nuget push ./bin/Release/DotNetLightning.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
+            dotnet nuget push ./bin/Release/DotNetLightning.Core.1*.nupkg -k ${{ secrets.NUGET_API_KEY }} -s https://api.nuget.org/v3/index.json
         fi
+      if: success()
```

Comment:
Are you sure this `if` is needed? if previous steps have failed, this step is already not run